modules/nf-core/gsea/gsea/main.nf

```diff
@@ -0,0 +1,76 @@
+    tuple val(meta), path("*/gsea_report_for_${meta.reference}*.html"),  path("*/gsea_report_for_${meta.target}*.html")                     , emit: report_htmls
+    tuple val(meta), path("*/ranked_gene_list*.tsv")                                                                                        , emit: ranked_gene_list
+    tuple val(meta), path("*/gene_set_sizes.tsv")                                                                                           , emit: gene_set_sizes
+    tuple val(meta), path("*/butterfly_plot.png")                                                                                           , emit: butterfly_plot
+    tuple val(meta), path("*/global_es_histogram.png")                                                                                      , emit: histogram
+    tuple val(meta), path("*/heat_map_1.png")                                                                                               , emit: heatmap
+    tuple val(meta), path("*/pvalues_vs_nes_plot.png")                                                                                      , emit: pvalues_vs_nes_plot
+    tuple val(meta), path("*/ranked_list_corr_2.png")                                                                                       , emit: ranked_list_corr
+    tuple val(meta), path("*/[!gene_set_size|gsea_report|ranked_gene_list]*.tsv")                                                           , emit: gene_set_tsv, optional: true
+    tuple val(meta), path("*/[!gsea_report|heat_map_corr_plot|index|pos_snapshot|neg_snapshot]*.html")                                      , emit: gene_set_html, optional: true
+    tuple val(meta), path("*/[!butterfly|enplot|global_es_histogram|gset_rnd_es_dist|heat_map|pvalues_vs_nes_plot|ranked_list_corr]*.png")  , emit: gene_set_heatmap, optional: true
+    tuple val(meta), path("*/*_snapshot*.html")                                                                                             , emit: snapshot, optional: true
+    tuple val(meta), path("*/enplot*.png")                                                                                                  , emit: gene_set_enplot, optional: true
+    tuple val(meta), path("*/gset_rnd_es_dist*.png")                                                                                        , emit: gene_set_dist, optional: true
+    tuple val(meta), path("*/*.zip")                                                                                                        , emit: archive, optional: true
+    path "versions.yml"                                                                                                                     , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    """
+    # Run GSEA
+    gsea-cli GSEA \
+        -res $gct \
+        -cls ${cls}#${meta.target}_versus_${meta.reference} \
```

Comment:
I agree with @nvnieuwk here! Is `-cls` mandatory here? If the tool fails if you dont provide it, you can just let the tool just that and the developer will learn ;) Otherwise you could just check that `-cls` exists in the `ext.args` as Nicolas suggests

Reply:
@nvnieuwk @jfy133 Thanks for the feedback.

The issue is, I'd really like to be able to refer to the treatment and reference levels in the output files (I want the receiving workflows to 'know' which files refer to which side of the contrast), which kind of predicates meta or a `val()', right? I can't refer to an args map there.

Note that CLS is a mandatory file input providing the classes, combined with that contrast definition :-)

Reply:
Or would the solution be to use static 'reference' and 'target' and rename the outputs?